### PR TITLE
Sourcery Starbot ⭐ refactored jamesgagg/comp_house

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>master</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/comp_house master
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: ch_mysql.py [hunk 1]
```diff
@@ -17,9 +17,11 @@ def __init__(self, config):
...
     def __str__(self):
         '''prints all attributes and their values'''
-        attributes = []
-        for key in self.__dict__:
-            attributes.append("{key}='{value}'".format(key=key, value=self.__dict__[key]))
+        attributes = [
+            "{key}='{value}'".format(key=key, value=self.__dict__[key])
+            for key in self.__dict__
+        ]
+
```
Comment (SourceryAI) [inline ch_mysql.py:24]: Function `MysqlConnection.__str__` refactored with the following changes:
- Convert for loop into list comprehension ([`list-comprehension`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#list-comprehension))

File: ch_mysql.py [hunk 2]
```diff
@@ -61,9 +63,11 @@ def __init__(self, mysql_connection):
...
```
Comment (SourceryAI) [inline ch_mysql.py:70]: Function `MysqlCursor.__str__` refactored with the following changes:
- Convert for loop into list comprehension ([`list-comprehension`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#list-comprehension))

File: ch_mysql.py [hunk 3]
```diff
@@ -93,8 +97,7 @@ def execute(self, query):
 def list_from_sql_query(mysql_cursor, query):
     '''return a list of the MySQL databases using the supplied cursor'''    
     mysql_cursor.execute(query)
-    mysql_list = [x[0] for x in mysql_cursor.fetchall()]
-    return mysql_list
+    return [x[0] for x in mysql_cursor.fetchall()]
```
Comment (SourceryAI) [inline ch_mysql.py:100]: Function `list_from_sql_query` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: ch_mysql.py [hunk 4]
```diff
@@ -128,12 +131,11 @@ def destroy_connection_cursor(cursor, connection):
 
 def mysqlconfig_from_config(config):
     '''creates a mysql compatible config from the program config'''
-    mysql_config = {
+    return {
         'user': config['mysql_user'],
         'password': config['mysql_password'],
         'host': config['mysql_host']
         }
-    return mysql_config
```
Comment (SourceryAI) [inline ch_mysql.py:136]: Function `mysqlconfig_from_config` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: ch_sql_queries_to_csv_example.py [hunk 5]
```diff
@@ -24,9 +24,11 @@ def sic_like(code_list):
...
     out_string = []
     for code in code_list:
-        temp_string = []
-        for i in range(1, 5):
-            temp_string.append('SICCode_SicText_' + str(i) + ' LIKE \'' + code + '%\'')
+        temp_string = [
+            'SICCode_SicText_' + str(i) + ' LIKE \'' + code + '%\''
+            for i in range(1, 5)
+        ]
+
```
Comment (SourceryAI) [inline ch_sql_queries_to_csv_example.py:31]: Function `sic_like` refactored with the following changes:
- Convert for loop into list comprehension ([`list-comprehension`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#list-comprehension))

File: ch_sql_queries_to_csv_example.py [hunk 6]
```diff
@@ -35,9 +37,11 @@ def sic_like(code_list):
...
 def postcodes_like(postcode_list):
     '''create mysql 'like' string for supplied postcode list'''
-    out_string = []
-    for postcode in postcode_list:
-        out_string.append('RegAddress_PostCode LIKE \'' + postcode + '%\'')
+    out_string = [
+        'RegAddress_PostCode LIKE \'' + postcode + '%\''
+        for postcode in postcode_list
+    ]
+
```
Comment (SourceryAI) [inline ch_sql_queries_to_csv_example.py:44]: Function `postcodes_like` refactored with the following changes:
- Convert for loop into list comprehension ([`list-comprehension`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#list-comprehension))

File: ch_tables.py [hunk 7]
```diff
@@ -23,14 +23,12 @@ def __init__(self, table_name, fields,
 
     def sql_create_table_str(self):
         '''create the string to pass to mysql to create the table'''
-        table_creation_string = self.sql_create_table_input_str.format(
+        return self.sql_create_table_input_str.format(
             *self.sql_create_table_input_format)
-        return table_creation_string
```
Comment (SourceryAI) [inline ch_tables.py:28]: Function `CHTable.sql_create_table_str` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: ch_tables.py [hunk 8]
```diff
@@ -23,14 +23,12 @@ def __init__(self, table_name, fields,
...
 
     def sql_insert_data_str(self, data):
         '''inserts data into the table'''
-        insert_string = self.sql_input_input_str.format(*self.sql_input_input_format + data)
-        return insert_string
+        return self.sql_input_input_str.format(*self.sql_input_input_format + data)
```
Comment (SourceryAI) [inline ch_tables.py:31]: Function `CHTable.sql_insert_data_str` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: ch_tables.py [hunk 9]
```diff
@@ -226,8 +224,7 @@ def get_config_from_file():
     config_tree = ET.parse('config.xml')
     config_root = config_tree.getroot()
 
-    config = {}
-    config['mysql_user'] = config_root.find('MySQL').find('user').text
+    config = {'mysql_user': config_root.find('MySQL').find('user').text}
```
Comment (SourceryAI) [inline ch_tables.py:227]: Function `get_config_from_file` refactored with the following changes:
- Merge dictionary assignment with declaration ([`merge-dict-assign`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#merge-dict-assign))

**Comment by jamesgagg** [comment]
Pull request is from a bot. Looking to manually update soon anyway.

**Pull request closed without merging**